### PR TITLE
Issue #10731 wrong context attribute name javax.servlet instead of jakarta.servlet

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <olamy@apache.org>

### CODE REVIEW

File: jetty-server/src/main/java/org/eclipse/jetty/server/MultiPartFormInputStream.java [hunk 1]
```diff
@@ -378,7 +378,7 @@ public MultiPartFormInputStream(InputStream in, String contentType, MultipartCon
      * @param in Request input stream
      * @param contentType Content-Type header
      * @param config MultipartConfigElement
-     * @param contextTmpDir javax.servlet.context.tempdir
+     * @param contextTmpDir jakarta.servlet.context.tempdir
```
Comment (sbordet) [inline jetty-server/src/main/java/org/eclipse/jetty/server/MultiPartFormInputStream.java]:
```suggestion
     * @param contextTmpDir {@value ServletContext#TEMPDIR}
```

File: jetty-server/src/main/java/org/eclipse/jetty/server/MultiPartInputStreamParser.java [hunk 2]
```diff
@@ -386,7 +386,7 @@ public MultiPartInputStreamParser(InputStream in, String contentType, MultipartC
      * @param in Request input stream
      * @param contentType Content-Type header
      * @param config MultipartConfigElement
-     * @param contextTmpDir javax.servlet.context.tempdir
+     * @param contextTmpDir jakarta.servlet.context.tempdir
```
Comment (sbordet) [inline jetty-server/src/main/java/org/eclipse/jetty/server/MultiPartInputStreamParser.java]: Ditto

File: jetty-server/src/main/java/org/eclipse/jetty/server/Request.java [hunk 3]
```diff
@@ -2424,12 +2424,12 @@ private MultiParts newMultiParts(MultipartConfigElement config, int maxParts) th
         {
             case RFC7578:
                 return new MultiParts.MultiPartsHttpParser(getInputStream(), getContentType(), config,
-                    (_context != null ? (File)_context.getAttribute("javax.servlet.context.tempdir") : null), this, maxParts);
+                    (_context != null ? (File)_context.getAttribute("jakarta.servlet.context.tempdir") : null), this, maxParts);
```
Comment (sbordet) [inline jetty-server/src/main/java/org/eclipse/jetty/server/Request.java]:
```suggestion
                    (_context != null ? (File)_context.getAttribute(ServletContext.TEMPDIR) : null), this, maxParts);
```

File: jetty-server/src/main/java/org/eclipse/jetty/server/Request.java [hunk 4]
```diff
@@ -2424,12 +2424,12 @@ private MultiParts newMultiParts(MultipartConfigElement config, int maxParts) th
...
 
             case LEGACY:
             default:
                 return new MultiParts.MultiPartsUtilParser(getInputStream(), getContentType(), config,
-                    (_context != null ? (File)_context.getAttribute("javax.servlet.context.tempdir") : null), this, maxParts);
+                    (_context != null ? (File)_context.getAttribute("jakarta.servlet.context.tempdir") : null), this, maxParts);
```
Comment (sbordet) [inline jetty-server/src/main/java/org/eclipse/jetty/server/Request.java]: Ditto.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-server/src/main/java/org/eclipse/jetty/server/MultiPartFormInputStream.java (hunk 1 above)
Comment (olamy) [reply jetty-server/src/main/java/org/eclipse/jetty/server/MultiPartFormInputStream.java]: done. wtih slight change using fqcn to fix javadoc issue

**Comment by olamy** [comment]
@sbordet good idea using the constant from `ServletContext` so I change to use that in few other places.

**Review by gregw — APPROVED** [review]

**Review by sbordet — APPROVED** [review]

**Pull request merged**